src/sage/graphs/orientations.py

```diff
@@ -41,6 +41,233 @@
 from sage.graphs.digraph import DiGraph
 
 
+def acyclic_orientations(G):
+    r"""
+    Return an iterator over all acyclic orientations of an undirected graph `G`.
+
+    ALGORITHM:
+
+    The algorithm is based on a paper by Mathew B. Squire, presenting an efficient
```

Comment:
Can you add a citation to the paper? (See the REFERENCES part of the [developer guide](https://doc.sagemath.org/html/en/developer/coding_basics.html#headings-of-sage-library-code-files)).